src/plugins/formulas/formulas.js

```diff
@@ -20,6 +27,59 @@ export class Formulas extends BasePlugin {
     return PLUGIN_PRIORITY;
   }
 
+  /**
+   * Plugin settings.
+   *
+   * @private
+   */
+  #settings = this.hot.getSettings()[PLUGIN_KEY];
+
+  /**
+   * Flag used to bypass hooks in internal operations.
+   *
+   * @private
+   * @type {boolean}
+   */
+  #internal = false;
```

Comment:
No idea for a better name, but `internal` doesn't say much at all, should change imo.